pypicloud/cache/dynamo.py

```diff
@@ -111,20 +199,99 @@ def configure(cls, settings):
         return kwargs
 
     def reload_from_storage(self):
-        pkgs = set()
-        for pkg in self.engine.scan(DynamoPackage):
-            pkgs.add(pkg.filename)
-
-        to_save = []
-        for pkg in self.storage.list(self.package_class):
-            if pkg.filename in pkgs:
-                pkgs.remove(pkg.filename)
-            else:
-                to_save.append(pkg)
-
-        self.engine.save(to_save, overwrite=True)
-
-        self.engine.delete_key(DynamoPackage, pkgs)
+        LOG.info("Recalculating cache.")
+
+        def normalize_last_mod(o):
+            """ Make it UTC without mized TZ comparisons """
+            # HACK Don't make flywheel compare mixed TZ-awareness objects.
+            d = o.last_modified.replace(tzinfo=UTC)
+            o.last_modified = None
+            o.last_modified = d
+            return o
```

Comment:
I think the correct solution here may be to either make everything else in pypicloud use a timezone-aware datetime, or to make flywheel play nice when changing a datetime to have/remove a timezone